src/main/java/ai/shreds/adapter/AdapterCategoryController.java

```diff
@@ -36,39 +35,29 @@ public AdapterCategoryController(ApplicationCreateCategoryInputPort createCatego
 
     @PostMapping
     public SharedCreateCategoryResponse createCategory(@Valid @RequestBody SharedCreateCategoryRequest request) {
-        validateCategoryData(request.getName(), request.getDescription());
         try {
             log.info("Creating category with name: {}", request.getName());
-            return new SharedCreateCategoryResponse(
-                createCategoryService.createCategory(request.getName(), request.getDescription()).getId(),
-                request.getName(),
-                request.getDescription()
-            );
+            return createCategoryService.createCategory(request.getName(), request.getDescription());
         } catch (IllegalArgumentException e) {
             log.error("Invalid input data: ", e);
-            throw new InvalidInputException("Invalid input data.");
+            throw new RuntimeException("Invalid input data.");
         } catch (Exception e) {
             log.error("Error creating category: ", e);
-            throw new ServerErrorException("Internal server error.");
+            throw new RuntimeException("Internal server error.");
         }
     }
 
     @PutMapping("/{id}")
     public SharedUpdateCategoryResponse updateCategory(@PathVariable String id, @Valid @RequestBody SharedUpdateCategoryRequest request) {
-        validateCategoryData(request.getName(), request.getDescription());
         try {
             log.info("Updating category with id: {}", id);
-            return new SharedUpdateCategoryResponse(
-                updateCategoryService.updateCategory(id, request.getName(), request.getDescription()).getId(),
-                request.getName(),
-                request.getDescription()
-            );
+            return updateCategoryService.updateCategory(id, request.getName(), request.getDescription());
```

Comment:
the method must return a SharedUpdateCategoryResponse instead of SharedCategoryDTO